inc/az_ulib_action_api.h

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+#ifndef AZ_ULIB_ACTION_API_H
+#define AZ_ULIB_ACTION_API_H
+
+#include "azure_macro_utils/macro_utils.h"
+#include "az_ulib_base.h"
+#include "ulib_result.h"
+#include "umock_c/umock_c_prod.h"
+
+#ifdef __cplusplus
+#include <stdint.h>
+extern "C" {
+#include <cstdint>
+#endif /* __cplusplus */
+
+/**
+ * @file    az_ulib_action_api.h
+ *
+ * @brief   Actions are properties, methods and events that can be handle by an interface.
+ */
+
+/**
+ * @brief Enumerator that defines the type of the action.
+ */
+MU_DEFINE_ENUM(
+    AZ_ULIB_ACTION_TYPE,
+    AZ_ULIB_ACTION_TYPE_PROPERTY      = 0x00, /**<Read and write property */
+    AZ_ULIB_ACTION_TYPE_METHOD        = 0x01, /**<Synchronous command that can be invoke by other 
+                                              modules in the system */
+    AZ_ULIB_ACTION_TYPE_METHOD_ASYNC  = 0x02, /**<Asynchronous task that can be invoke by other
+                                              modules in the system */
+    AZ_ULIB_ACTION_TYPE_EVENT         = 0x03  /**<Event that other modules in the system can 
+                                              subscribe to be notified */
+);
+
+#define AZ_ULIB_ACTION_TYPE_BITS 0x03
+
+/**
+ * @brief       Token that unique identify the action.
```

Comment:
>unique identify [](start = 27, length = 15)

nit: uniquely identifies

Reply:
still not correct

---
In reply to: [338164953](https://github.com/Azure/azure-ulib-c/pull/40#discussion_r338164953) [](ancestors = 338164953)